addons/rmsmartshape/normal_range.gd

```diff
@@ -1,117 +1,95 @@
 @tool
 extends Resource
 class_name SS2D_NormalRange
-"""
-This class will determine if the normal of a vector falls within the specifed angle ranges
-- if begin and end are equal, any angle is considered to be within range
-- 360.0 and 0.0 degrees are considered equivilent
-"""
 
-@export_range (0, 360, 0) var begin: int = 0 : set = set_begin
-@export_range (0, 360, 0) var distance: int = 0 : set = set_distance
+## This class will determine if the normal of a vector falls within the specifed angle ranges.
+##
+## - if begin and end are equal, any angle is considered to be within range [br]
+## - 360.0 and 0.0 degrees are considered equivilent [br]
 
-# Deprecated variable
-var end = 0.0 : set = set_end
+@export_range (0, 360, 1) var begin: float = 0.0 : set = set_begin
+@export_range (0, 360, 1) var distance: float = 0.0 : set = set_distance
 
 # This is a hack to support the custom editor, needed a property
 # to exist to lock the TextureProgressBar to.  Makes it flow better
 # in the Inspector.
 @export var edgeRendering: Vector2
 
 
-func set_distance(f: float):
+func set_distance(f: float) -> void:
 	distance = f
```

Comment:
I just added the return type... You probably mean the setter problem I found in some other places. It usually looks like this:
```
var prop: int:
    set(value):
        set_prop(value)

func set_prop(value: int):
    prop = value  ## this will cause infinite recursion, since setting prop triggers original setter
```

Reply:
It's fine.
![Screenshot from 2023-03-12 01-13-05](https://user-images.githubusercontent.com/2766569/224517239-9fef346a-3d31-4c04-b64f-f11aefbf937b.png)

Reply:
You're right, I just tested it. I also just checked my [alpha branch](https://github.com/mphe/SmartShape2D/blob/b2dee93ec7b9025ac1a4abc2d04bfd2009e20943/addons/rmsmartshape/normal_range.gd#L11) because I really remembered that there was something causing recursion, but it was because I kept `end` for compatibility.